### ISSUE TITLE
installing radarr in freenas jail

### ISSUE THREAD

**Issue by mfscorp** [Radarr/Radarr#2324]
Hi,
Im fairly new to freenas and Linux but pretty wel self learning on the go:
i have freenas 11.0.u4 installed and followed this guide of installing radarr:
https://github.com/Radarr/Radarr/wiki/Installation
also added the script from down below at etc/rc.d/radarr
But after restarting the jail and navigating to the ip:7878 i do not get any web
I have chowned all to media:media from jail/radarr/ and also set all perms to 775 of jail/radar folder.
But still no success

Can anyone help me solve this issue?

If i need to provide something please explain me exactly what i have to do also with commandlines (if needed) as i said im fairly new to this but pretty quick learner 

regards

**Comment by InnerSpark** [comment]
having the same issues. Looks like it is something to do with fsync and not allowing the service to start.

`service radarr start
[Info] Bootstrap: Starting Radarr - /usr/local/Radarr/Radarr.exe - Version 0.2.0.870
[Info] MigrationLogger: *** Migrating data source=/.config/Radarr/nzbdrone.db;cache size=-10485760;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 ***
/usr/local/lib/libsqlite3.so: Undefined symbol "fdatasync"`

**Comment by mannyAh-dy** [comment]
i am having the same exact issue. the radarr service is not starting,

**Comment by mikedebian** [comment]
/usr/local/lib/libsqlite3.so: Undefined symbol "fdatasync"

Same issue, FreeNAS 11.0.u4

**Comment by mfscorp** [comment]
https://forums.freenas.org/index.php?threads/installing-jail-with-radarr.59599/

had an answer that there is a pkg radarr on the official repos but that also did not work.

**Comment by mikedebian** [comment]
This bug also affects Sonarr.

**Comment by jj-uk** [comment]
Yep - Radarr and Sonarr are both down due to this bug. Uuugh!

**Comment by galli-leo** [comment]
Maybe a too old / new version of libsqlite? Seems like a function is missing which we are using / linked against?

**Comment by galli-leo** [comment]
See: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=221672 you have to install 11.1

**Comment by jj-uk** [comment]
FreeNAS is running on 11.0, there is no 11.1 yet.

This is a bug as you're broken Radarr and Sonarr for ALL freenas users!

**Comment by Wolfeman0101** [comment]
I'm on 11.1 and I can't start Radarr

**Comment by mfscorp** [comment]
i also installed freenas 11.1 but still have the same issue with new fresh jail.
I cant get on the website of radarr with ip:7878

**Comment by jj-uk** [comment]
In the jail, what does: freebsd-version show?

**Comment by mfscorp** [comment]
were can i see that in the jail?
i go to folder /jails/Radarr/ and then?

**Comment by jj-uk** [comment]
No. Go into the jail. At the shell type:

freebsd-version

**Comment by mfscorp** [comment]
11.0-RELEASE-p1
:\

**Comment by jj-uk** [comment]
When you log into freenas,  click use beta ui on the login screen. The beta ui creates and works with a new jail type using 'iocage'. The old ui deals with only 'warden' jails.

Create a jail and select 11.1 from the drop list.  Also select allow-raw-sockets and vnet/vimage.
Then create the jail.

Then via ssh log into freenas. Then into the jail like this:

# iocage console <jailname>

# freebsd-version

Then try to ping the real world;

# ping bbc.co.uk

https://doc.freenas.org/11/jails.html
Scroll to last section.

**Comment by mfscorp** [comment]
ok ping did not work.
Ill do some research and see how to set this up correctly
iocage is new for me.
The old "warden" setup was pretty easy to do..

**Comment by jj-uk** [comment]
Please make notes of what you do are post back!

**Comment by mfscorp** [comment]
will post here onces i got it working..
but dun forget im fairly new to all of this ...
google / these kind of forums are my best friends ;)

**Comment by jj-uk** [comment]
I now have this working with a standard jail (not an iocage jail).

Create a jail using the OLD gui, then:

 DO NOT start jail until all is done, if you do, delete /tmp/radarr_* folders.

>pkg install radarr
>
>portsnap fetch
>portsnap extract
>cd /usr/ports/devel/readline
>make deinstall
>make install clean ALLOW_UNSUPPORTED_SYSTEM=true
>cd /usr/ports/databases/sqlite3
>make deinstall
>make install clean ALLOW_UNSUPPORTED_SYSTEM=true
>
>rmuser media
>pw useradd -n media -u 816 -d /nonexistent -s /usr/sbin/nologin
>chown -R media:media /usr/local/radarr
>chown -R media:media /usr/local/share/radarr
>sysrc radarr_enable=YES
>sysrc radarr_user=media
>
> restart jail. Radarr is available on <jail id>:7878
>

**Comment by Barbait** [comment]
I had a problem with the steps right above. I followed everything in the first block.
I have modified something in the second block, why? Because i needed group ( daemons group) write permissions to a folder and i got problem with the media user.

`rmuser media
pw useradd -n radarrdaemon -u 816 -d /nonexistent -s /usr/sbin/nologin
chown -R radarrdaemon:daemons /usr/local/radarr
chown -R radarrdaemon:daemons /usr/local/share/radarr
sysrc radarr_enable=YES
sysrc radarr_user=radarrdaemon
`
Outside the jail i have a group (daemons) with matching GID with the daemons group inside the jail, same for the radarrdaemon user.
Don't forget to add the user to the group inside and outside the jail.

Hope it helps someone.
